source/plugins/go/src/extension/extension.go

```diff
@@ -135,47 +143,87 @@ func getDataCollectionSettingsInterface() (map[string]interface{}, error) {
 	return dataCollectionSettings, nil
 }
 
-func getDataTypeToStreamIdMapping(hasNamedPipe bool) (map[string]string, error) {
+func getDataTypeToStreamIdMappingFromCIExtension(hasNamedPipe bool) (map[string]string, error) {
 	datatypeOutputStreamMap := make(map[string]string)
-
-	extensionConfigs, err := getExtensionConfigs()
+	extensionData, err := getExtensionData(ContainerInsightsExtension, ContainerInsightsExtensionVersion)
 	if err != nil {
 		return datatypeOutputStreamMap, err
 	}
+	extensionConfigs := extensionData.ExtensionConfigs
 	outputStreamDefinitions := make(map[string]StreamDefinition)
 	if hasNamedPipe == true {
-		extensionData, err := getExtensionData()
-		if err != nil {
-			return datatypeOutputStreamMap, err
-		}
 		outputStreamDefinitions = extensionData.OutputStreamDefinitions
 	}
 	for _, extensionConfig := range extensionConfigs {
 		outputStreams := extensionConfig.OutputStreams
 		for dataType, outputStreamID := range outputStreams {
-				if hasNamedPipe {
-					datatypeOutputStreamMap[dataType] = outputStreamDefinitions[outputStreamID.(string)].NamedPipe
-				} else {
-					datatypeOutputStreamMap[dataType] = outputStreamID.(string)
-				}
+			if hasNamedPipe {
+				datatypeOutputStreamMap[dataType] = outputStreamDefinitions[outputStreamID.(string)].NamedPipe
+			} else {
+				datatypeOutputStreamMap[dataType] = outputStreamID.(string)
+			}
 		}
 	}
 	return datatypeOutputStreamMap, nil
 }
 
+func getNamespacesFromDataCollectionSettings(dataCollectionSettings map[string]interface{}) []string {
+	var namespaces []string
+	if len(dataCollectionSettings) > 0 {
+		namespacesSetting, found := dataCollectionSettings[EXTENSION_SETTINGS_DATA_COLLECTION_SETTINGS_NAMESPACES].([]interface{})
+		if found {
+			if len(namespacesSetting) > 0 {
+				// Remove duplicates from the namespacesSetting slice
+				uniqueNamespaces := make(map[string]bool)
+				for _, ns := range namespacesSetting {
+					str, ok := ns.(string)

```

Comment:
if str, ok := ns.(string); ok {
				uniqueNamespaces[strings.ToLower(str)] = true
			} else {
				logger.Println("ExtensionUtils::getNamespacesForDataCollection: namespace:", ns, "not valid hence skipping")
			}